pkg/operation/botanist/secrets.go

```diff
@@ -899,21 +899,11 @@ func computeProjectSecretName(shootName, suffix string) string {
 	return fmt.Sprintf("%s.%s", shootName, suffix)
 }
 
-// SyncShootCredentialsToGarden copies the kubeconfig generated for the user as well as the SSH keypair to
-// the project namespace in the Garden cluster. If control plane monitoring is enabled, then the
-// monitoring credentials for the user-facing monitoring stack are also copied.
+// SyncShootCredentialsToGarden copies the kubeconfig generated for the user, the SSH keypair to
+// the project namespace in the Garden cluster and the monitoring credentials for the
+// user-facing monitoring stack are also copied.
 func (b *Botanist) SyncShootCredentialsToGarden() error {
-	projectSecrets := map[string]string{secretSuffixKubeConfig: "kubecfg", secretSuffixSSHKeyPair: gardencorev1alpha1.SecretNameSSHKeyPair}
-
-	if b.Shoot.WantsControlPlaneMonitoring {
-		projectSecrets[secretSuffixMonitoring] = "monitoring-ingress-credentials-users"
-	} else {
-		if err := b.K8sGardenClient.DeleteSecret(b.Shoot.Info.Namespace, computeProjectSecretName(b.Shoot.Info.Name, secretSuffixMonitoring)); err != nil && !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-
-	for key, value := range projectSecrets {
+	for key, value := range map[string]string{secretSuffixKubeConfig: "kubecfg", secretSuffixSSHKeyPair: gardencorev1alpha1.SecretNameSSHKeyPair, secretSuffixMonitoring: "monitoring-ingress-credentials-users"} {
```

Comment:
I would still keep projectSecrets, if it gets bigger the loop becomes hard to digest.